src/CommonAccessors.jl

```diff
@@ -0,0 +1,108 @@
+# Common get and set methods
+
+# NOTE this could be reduced with macros and function generation to even less code.
+
+export getLabel, label
+export getTags, setTags!, tags
+export getTimestamp, setTimestamp!, timestamp
+export getInternalId, interalId
+
+
+const DataLevel0 = Union{VariableDataLevel0, FactorDataLevel0}
+const DataLevel1 = Union{VariableDataLevel1, FactorDataLevel1}
+const DataLevel2 = Union{VariableDataLevel2, FactorDataLevel2}
+
+
+"""
+$SIGNATURES
+
+Return the label for a variable or factor.
+"""
+getLabel(v::DataLevel0) = v.label
+
+"""
+$SIGNATURES
+
+Return the label for a variable or factor.
+
+DEPRECATED label -> getLabel
+"""
+function label(v::DataLevel0)
+  @warn "Deprecated label, use getLabel instead."
+  getLabel(v)
+end
+
+
+"""
+$SIGNATURES
+
+Return the tags for a variable.
+"""
+getTags(v::DataLevel0) = v.tags
+
+
+"""
+$SIGNATURES
+
+Return the tags for a variable.
+
+DEPRECATED, tags -> getTags
+"""
+function tags(v::DataLevel0)
+  @warn "tags deprecated, use getTags instead"
+  getTags
+end
+
+"""
+$SIGNATURES
+
+Set the tags for a factor.
+"""
+setTags!(f::DataLevel0, tags::Vector{Symbol}) = f.tags = tags
```

Comment:
This will not work on skeleton and depending on the final vote on #182 DFGVariable. 
I also think its a bad idea to allow memory pointers to be changed.

Reply:
> This will not work on skeleton

Got it thanks, so we can increase to `DataLevel1`?

> bad idea to allow memory pointers to be changed

So you'd suggest we rather use something like this?
```julia
function setTags!(f::DataLevel1, tags::Vector{Symbol})
  vtags = getTags(f)
  todelete = setdiff(vtags, tags)
  union!(vtags, tags)
  for td in todelete
    for ind in findall(x->x==td, vtags)
      delete!(vtags, ind..)
    end
  end
end
```

Reply:
Tags is in skeleton, but its not mutable. So you have to work with the original memory. Its not mutable for performance.

Reply:
understood thanks, the suggested function here should work with the existing pointer -- i.e. with an immutable struct.

Reply:
with variable order in SkeletonFactor, I just did this:
```julia
resize!(factor._variableOrderSymbols, length(variableLabels))
factor._variableOrderSymbols .= variableLabels
```
so something like this seems a bit easier to read?
```julia
resize!(f.tags, length(tags))
f.tags .= tags
```

Reply:
That's if you want to replace all of them.

Reply:
Just changed it out to that for now.